src/SonarScanner.MSBuild.PreProcessor/AnalysisConfigGenerator.cs

```diff
@@ -104,32 +97,46 @@ public static AnalysisConfig GenerateFile(ProcessedArgs localSettings,
             config.SetSettingsFilePath(localSettings.PropertiesFileName);
         }
 
-        HandleCoverageExclusions(config, localSettings, serverProperties);
+        CoverageExclusions.UpdateConfig(config, localSettings, serverProperties);
         config.Save(buildSettings.AnalysisConfigFilePath);
         return config;
     }
 
+    private static void AddSetting(AnalysisProperties properties, string id, string value)
+    {
+        var property = new Property(id, value);
+
+        // Ensure it isn't possible to write sensitive data to the config file
+        if (!property.ContainsSensitiveData())
+        {
+            properties.Add(new(id, value));
+        }
+    }
+
     // See https://sonarsource.atlassian.net/browse/SCAN4NET-29
-    // This method is a hack and should be removed when we properly support excluding coverage files in the scanner-engine (https://sonarsource.atlassian.net/browse/SCANENGINE-18).
+    // This class is a hack and should be removed when we properly support excluding coverage files in the scanner-engine (https://sonarsource.atlassian.net/browse/SCANENGINE-18).
     // The idea is that we are manually adding the coverage paths to the exclusions, so that they do not appear on the analysis.
-    private static void HandleCoverageExclusions(AnalysisConfig config, ProcessedArgs localSettings, IDictionary<string, string> serverProperties)
+    private static class CoverageExclusions
```

Comment:
I like that it's a class now. What do you think about:
- extracting it outside of the parent class
- move it to another file
- rename it to coverage exclusions processor and make it an instance type instead of static
- passing it to the constructor

Reply:
I think I prefer to leave it private in here, mainly for the following reasons:

- This class is tightly coupled with how we generate the AnalysisConfig
- The test scaffolding needed is more or less the same for `GenerateFile` and `UpdateConfig`